requests/status_codes.py

```diff
@@ -114,6 +114,7 @@ def doc(code):
 
     global __doc__
     __doc__ = (__doc__ + '\n' +
```

Comment:
So let's look at how this evaluates:

```
__doc__ = (None + '\n' + None)
```

In your case. Do we instead want

```py
__doc__ = (__doc__ + '\n' +
           '\n'.join(doc(code) for code in sorted(_codes))) if __doc__ is not None else None
```

Which checks if  `__doc__` is `None` prior to trying to evaluate the parenthetical?

Reply:
```
(x + '\n' + 'code_stuff'  if x is not None else None)
```
evaluates to `None` when `x` is `None` - the addition is not evaluated.

Reply:
Yep, I don't think we need to worry about evaluation unless the conditional is met. This looks good from my end unless @sigmavirus24 has further comments. Thanks for the quick follow up on this @martindurant!